app/uk/gov/hmrc/eoricommoncomponent/frontend/forms/models/mappings/LocalDateFormatter.scala

```diff
@@ -61,16 +63,20 @@ class LocalDateFormatter(emptyKey: String, invalidKey: String, args: Seq[String]
 
     fields.count(_._2.isDefined) match {
       case 3 =>
-        formatDate(key, data).left.map {
-          _.map(_.copy(key = key, args = args))
+        formatDate(key, data).left.map { leftValue =>
+          leftValue.map(formError => formError)
```

Comment:
Previously, we were always setting the key to "date-of-establishment" or "date-of-birth" now, if we have a form validation error we pass that through which contains the correct key to link the error message to the problematic field